### PR TITLE
Api 337: migration script to add "user_to_notify" and "is_user_authenticated" in job instance configuration

### PR DESCRIPTION
**Pull request by ahocquard**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**
Add "user_to_notify" key and "is_user_authenticated" key in the job instances's configuration

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: upgrades/1.7-2.0/Version_2_0_20170925181205_add_default_parameters_in_job_instances.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+ * @author    Alexandre Hocquard <alexandre.hocquard@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class Version_2_0_20170925181205_add_default_parameters_in_job_instances extends AbstractMigration implements ContainerAwareInterface
+{
+    use ContainerAwareTrait;
+
+    protected const JOB_INSTANCE_CODES = [
+        'add_product_value',
```
Comment (jjanvier) [inline upgrades/1.7-2.0/Version_2_0_20170925181205_add_default_parameters_in_job_instances.php]: why do change only those jobs and not all?

Comment (jjanvier) [reply upgrades/1.7-2.0/Version_2_0_20170925181205_add_default_parameters_in_job_instances.php]: what if the project contains other jobs?

Comment (ahocquard) [reply upgrades/1.7-2.0/Version_2_0_20170925181205_add_default_parameters_in_job_instances.php]: And that's the problem: what if the job has its own constraint?

**Comment by ahocquard** [comment]
Not needed for compute_product_models_descendants as the constraint does not contain the key.

**Pull request merged**